activerecord/lib/active_record/relation/batches.rb

```diff
@@ -201,15 +208,15 @@ def find_in_batches(start: nil, finish: nil, batch_size: 1000, error_on_ignore:
     #
     # NOTE: By its nature, batch processing is subject to race conditions if
     # other processes are modifying the database.
-    def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore: nil, order: :asc)
+    def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore: nil, order: :asc, use_ranges: nil)
```

Comment:
What about to make it more generic (extendable) in the future? I'm still trying to find out way to introduce PG cursors to ActiveRecord. This could be the way. What about `scope_by: :ids` by default and `scope_by: :range` with this optional addition? I can later try to contribute `scope_by: :cursor`.

:thinking: _I'm really bad at naming, I believe there is better name for this argument._

Reply:
Do I correctly understand, that cursors can be used only when `load: true` is passed?

Reply:
No, why? You can move in iteration using `MOVE` when you don't ask for the data to be loaded.  And even it would be needed, what's the problem?